kratos/solving_strategies/schemes/residual_based_bdf_scheme.h

```diff
@@ -226,30 +225,30 @@ class ResidualBasedBDFScheme
     /**
      * @brief It initializes time step solution. Only for reasons if the time step solution is restarted
      * @param rModelPart The model of the problem to solve
-     * @param A LHS matrix
-     * @param Dx Incremental update of primary variables
-     * @param b RHS Vector
+     * @param rA LHS matrix
+     * @param rDx Incremental update of primary variables
+     * @param rb RHS Vector
      * @todo I cannot find the formula for the higher orders with variable time step. I tried to deduce by myself but the result was very unstable
      */
-
     void InitializeSolutionStep(
         ModelPart& rModelPart,
-        TSystemMatrixType& A,
-        TSystemVectorType& Dx,
-        TSystemVectorType& b
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb
         ) override
     {
         KRATOS_TRY;
 
         ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
+        ProcessInfo& r_previous_process_info = r_current_process_info.GetPreviousTimeStepInfo(1);
 
-        ImplicitBaseType::InitializeSolutionStep(rModelPart, A, Dx, b);
+        ImplicitBaseType::InitializeSolutionStep(rModelPart, rA, rDx, rb);
 
         const double delta_time = r_current_process_info[DELTA_TIME];
-        const double previous_delta_time = r_current_process_info.GetPreviousTimeStepInfo(1)[DELTA_TIME];
+        const double previous_delta_time = std::abs(r_previous_process_info[DELTA_TIME]) > ZeroTolerance ? r_previous_process_info[DELTA_TIME] : delta_time;
```

Comment:
I don't like too much this check. I think that the problem is in how the very first `CloneTimeStep` calls are done to fill the buffer before the very first `InitializeSolutionStep` call. If this is done in accordance with the required buffer size, this check shouldn't be needed.

Actually I think it is a good idea to keep the error as it is. In any case, please correct me if I'm wrong.

Reply:
Well, what is clear is that if the previous time step is equal to zero means that it has not been initialized properly. This can be solved apart, but that change will take long to be solved, as it implies a lot of discussion on the @KratosMultiphysics/structural-mechanics side. Normally we don't fill the buffer in the same way it is done on the fluids.

Meanwhile this will solve the issue, and what is doing is not wrong

Reply:
Now you raised another important discussion, to unify the way the buffers are filled, since this becomes in a problem when coupling both applications.

Aside of this, I agree that this circumvents (not solves) the problem right now. Having this in mind, I'd rather start the discussion than adding a (temporary) solution that silently hides the source of the problem.

Reply:
I assure you it will be a very long discussion, thing that most of structural problems are static, not dynamic. The need of filling the buffer is completely different. I would approve this, the "problem" that is "hiding" is several orders of magnitude more complex that the current solution. This will not make any harm, and it will continue working after the fix that will take several months (probably) to arrive

Reply:
Up to @KratosMultiphysics/structural-mechanics group then. But take into account that this is scheme is part of the Kratos core.

Reply:
I discussed the issue of how to fill the buffer several times already with @jcotela 
Thing is that I am not sure if it can be unified. StructuralMechanics has to fill it ["backwards in time"](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py#L340-L361) because the problem can be static and hence have only one step.

For Fluids the solving only starts after the buffer is filled, apparently this is needed to applying the BCs / initial conditions correctly

=> long story short, I am not sure if we can find a common solution due to different technical requirements

Reply:
That's basically what I tried to explain, it is a long discussion